### PR TITLE
Improved README

### PR DESCRIPTION
**Pull request by vfdev-5**
Description:
- Better selling readme ;)

cc @ykumards @sdesrozis

### CODE REVIEW

**Comment by vfdev-5** [comment]
> we should notice that (thaks to python) data are easly accessible inside function
👍 

> the use case of custom event is not clear here. User should call function direclty.
Maybe I just put a link on TTBPT

> Note for later : we must do a tutorial to explain how develop metrics :)

this is done [here](https://pytorch.org/ignite/master/metrics.html#how-to-create-a-custom-metric), right ?

File: README.md [hunk 1]
```diff
@@ -107,8 +107,84 @@ level of abstraction allows for a great deal more of flexibility, such
...
+        trainer.fire_event(CustomEvents.FOO_EVENT)
+```
+
+
+## Out-of-the-box metrics
+
+- Metrics for various tasks: Precision, Recall, Accuracy, Confusion Matrix, 
+IoU etc, ~20 regression metrics.
+
+- Users can compose its own metric by using basic arithmetical operations or torch methods:
```
Comment (ykumards) [inline README.md]: > Users can compose its own metric by using basic arithmetical operations or torch methods:

Shall we rephrase this as "Users can also compose their own metrics with ease from existing ones using arithmetic operations or torch methods" ?

File: README.md [hunk 2]
```diff
@@ -107,8 +107,121 @@ level of abstraction allows for a great deal more of flexibility, such
...
 multiple losses and metrics in your training loop.
 
-Ignite also allows for multiple handlers to be attached to events, and a
-finer granularity of events in the engine loop.
+## Power of Events & Handlers
+
+The cool thing with handlers (vs “callback” interfaces) is that it can be any function 
+with the correct signature (we only require the first argument to be engine), e.g. lambda, simple 
+function, class method etc. We do not require to inherit from an interface and override possibly 
+its abstract methods.
```
Comment (ykumards) [inline README.md]: Suggested edit:
The cool thing with handlers is that they offer unparalleled flexibility (compared to say, callbacks). Handlers can be any function (e.g. lambda, simple function, class method etc.) with the correct signature, we only require that the first argument be `engine`. Thus, we do not require to inherit from an interface and override its abstract methods which could unnecessarily bulk up your code and its complexity.

Comment (vfdev-5) [reply README.md]: Thanks Yogesh! Will update
